chrome/content/brief-overlay.js

```diff
@@ -192,12 +194,14 @@ var Brief = {
             Brief.constructTooltip();
     },
 
-    updateStatus: function Brief_updateStatus() {
-        if (!Brief.toolbarbutton)
+    updateStatus: function Brief_updateStatus(toolbarbutton) {
+        if (toolbarbutton == undefined)
+            toolbarbutton = Brief.toolbarbutton.forWindow(window);
```

Comment:
I think that the brief-overlay.js is loaded into every single window. That is it's instantiated for every window. Thus the updateStatus function inside the global Brief variable has access to the current window object.
I had to move the registration of the brief button to it's own jsm to ensure it is only run once per Firefox instance. The code in the brief-overlay.js should run separately for each window.